src/Simulation/Simulators.Tests/RuntimeMetadataTests.cs

```diff
@@ -186,159 +186,222 @@ public class IntrinsicTests
         [Fact]
         public void CNOT()
         {
-            var control = new FreeQubit(1);
-            var target = new FreeQubit(0);
-            var op = new QuantumSimulator().Get<Intrinsic.CNOT>();
-            var args = op.__DataIn__((control, target));
-            var expected = new RuntimeMetadata()
-            {
-                Label = "X",
-                FormattedNonQubitArgs = "",
-                IsAdjoint = false,
-                IsControlled = true,
-                IsMeasurement = false,
-                IsComposite = false,
-                Children = null,
-                Controls = new List<Qubit>() { control },
-                Targets = new List<Qubit>() { target },
+            var simulators = new CommonNativeSimulator[] { 
```

Comment:
An alternate approach may be to refactor unit tests to be theories instead of facts, and then to use different simulator types as theory data. Looking, the blog post at https://andrewlock.net/creating-strongly-typed-xunit-theory-test-data-with-theorydata/ seems to be a good summary of using `[ClassData]` and `TheoryData<T>` together to make strongly typed XUnit theories.

Reply:
Thank you for sharing this knowledge! Good posts.
If I get to refactoring the tests, then I will definitely consider making them Theories. But I don't promise to do that in this PR.